analyzers/tests/SonarAnalyzer.UnitTest/Helpers/KnownTypeTest.cs

```diff
@@ -0,0 +1,101 @@
+﻿/*
+ * SonarAnalyzer for .NET
+ * Copyright (C) 2015-2022 SonarSource SA
+ * mailto: contact AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.VisualBasic.Syntax;
+using SonarAnalyzer.Common;
+
+namespace SonarAnalyzer.UnitTest.Helpers;
+
+[TestClass]
+public class KnownTypeTest
+{
+    [TestMethod]
+    public void Matches_TypeSymbolIsNull_ThrowsArgumentNullException() =>
+        Assert.ThrowsException<ArgumentNullException>(() => new KnownType.RegularKnownType("typeName").Matches(null));
+
+    [DataTestMethod]
+    [DataRow("System.Action", true, "System", "Action", false)]
+    [DataRow("System.DateTime", false, "System", "Action", false)]
+    [DataRow("System.Collections.ArrayList", true, "System.Collections", "ArrayList", false)]
+    [DataRow("System.Threading.Timer", false, "System.Timers", "Timer", false)]
+    [DataRow("string[]", true, "", "String", true)]
+    [DataRow("System.String[]", true, "", "String", true)]
+    [DataRow("byte[]", true, "", "Byte", true)]
+    [DataRow("System.Byte[]", true, "", "Byte", true)]
+    [DataRow("System.Action<T>", false, "System", "Action", true)]
+    [DataRow("System.Action<T>", true, "System", "Action", true, "T")]
+    [DataRow("System.Action<T1>", false, "System", "Action", true, "T2")]
+    [DataRow("System.Action<T1, T2>", true, "System", "Action", true, "T1", "T2")]
+    [DataRow("System.Action<T1, T2>", false, "System", "Action", true, "T2", "T1")]
```

Comment:
I want to replicate the `RegularKnownType` ctor parameters. I think we will not have to modify this a lot and I would like to avoid the splitting logic from the test.